colossalai/utils/activation_checkpoint.py

```diff
@@ -136,14 +138,122 @@ def backward(ctx, *args):
         return (None, None) + grads
 
 
-def checkpoint(function, activation_offload, *args):
+def checkpoint(function, activation_offload, *args, use_reentrant: bool = True):
```

Comment:
You should test on densenet121 where bugs were reported in #1451.

Reply:
I have done some new tests on use_reentrant=False to test if it could solve the problem in report in #1451, I ran the test in #1451 locally and it passed, you could do some final check on the code. I will change the codegen after this PR is merged

Reply:
I will later add more logic for ckpt_solver. Approve~